addons/mrp/models/mrp_abstract_workorder.py

```diff
@@ -163,7 +163,9 @@ def _update_workorder_lines(self):
                     for values in self._generate_lines_values(move, qty_todo):
                         line_values['to_create'].append(values)
         # wo lines without move_id should also be deleted
-        for wo_line in self._workorder_line_ids().filtered(lambda w: not w.move_id and (not w.finished_workorder_id or w.product_id != w.finished_workorder_id.product_id)):
+        for wo_line in self._workorder_line_ids().filtered(
+            lambda w: not w.move_id and hasattr(w, 'finished_workorder_id') and (not w.finished_workorder_id or w.product_id != w.finished_workorder_id.product_id)
```

Comment:
It's correct but I prefere 
```python
lambda w: not w.move_id and w._get_finished_workoder_inverse_name() == 'finished_workorder_id' and (not w.finished_workorder_id or w.product_id != w.finished_workorder_id.product_id)
```
It's more 'Odoo compliant'

Or even better overide _update_workorder_lines in mrp_workorder.py (it should not be too difficult as it's a patch at the end of the method) to only work on `self` that are workorders.

Reply:
Thanks for the feedback.
I've moved the code in `mrp_workorder.py`